cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -163,6 +169,25 @@ func (p *BicepProvider) EnsureEnv(ctx context.Context) error {
 	}
 
 	if scope == azure.DeploymentScopeResourceGroup {
+		rgEval, err := rg.Envsubst(p.env.Getenv)
+		if err != nil {
+			return err
+		}
+
+		if rgEval != "" {
+			// TODO(ellismg): We set this here because the rest of the system reads from `.env` when it needs
+			// AZURE_RESOURCE_GROUP. But I'm not sure this is exactly what we want to be doing here - it feels wrong
+			// on some level that changing the value in `azure.yaml` ends up replacing what was in the `.env` file. It seems
+			// like if it is explicitly set in `azure.yaml` than we do not need to store it in the `.env` file at all.
+			//
+			// We also need to understand the policy of what happens if both are set - which one wins?  I think that perhaps
+			// it should be `AZURE_RESOURCE_GROUP` since that also allows overriding via `AZURE_RESOURCE_GROUP=foo azd up`.
+			p.env.DotenvSet(environment.ResourceGroupEnvVarName, rgEval)
```

Comment:
This is pretty gross, IMHO, but the rest of the system right now reads `environment.ResourceGroupEnvVarName` from the `.env` to find the resource group to use.  I suspect that I need to do a deeper refactoring such that we can consider both what is in the `azure.yaml` file and what is in the `.env` but right now doing things this way lets the rest of the system work.

But there are open questions, hence the big TODO and the draft nature of this PR...